ui/src/app/settings/views/Scripts/ScriptsUpload/ScriptsUpload.js

```diff
@@ -41,33 +41,24 @@ const ScriptsUpload = ({ type }) => {
   }, [savedScript, hasErrors, errors, dispatch]);
 
   const onDrop = useCallback(
-    (acceptedFiles, rejectedFiles) => {
-      if (acceptedFiles.length > 1 || rejectedFiles.length > 1) {
-        dispatch(
-          messages.add(`Only a single file may be uploaded.`, "caution")
-        );
-        return;
-      }
+    (acceptedFiles, fileRejections) => {
+      fileRejections.forEach((rejection) => {
+        rejection.errors.forEach((error) => {
+          if (error.code === "too-many-files") {
+            dispatch(
+              messages.add(`Only a single file may be uploaded.`, "caution")
+            );
```

Comment:
The built in error for this rejection isn't quite as nice (it just says "Too many files"), so preserving our custom error message here.

Reply:
I'm not sure if this is fixable, but when I dropped a few files I got this error for each file...